tests/unit/lms/events/subscriber_test.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+from h_matchers import Any
+
+from lms.events import FilesDiscoveredEvent
+from lms.events.subscriber import files_discovered
+from lms.models import File
+from tests import factories
+
+
+class TestFilesDiscovered:
```

Comment:
It looks like we're making life a lot harder for ourselves than we have to here, as we could just mock out the bulk object and make assertions about that but...

* We don't mock basic DB functionality, which this is now acting as
* This is the only place that actually brings together the bulk upsert, File values etc and makes sure they all work together

That said it's still super long winded for a very short piece of code and I'd appreciate any suggestions on how to cut it down.